controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -406,20 +411,43 @@ def test_get_object_by_id_volume_success(self):
         volume = self.svc.get_object_by_id("volume_id", "volume")
         self.assertEqual(volume.name, "volume_id")
 
-    def _prepare_mocks_for_create_snapshot(self, support_deduplicated_copy=True, source_has_deduplicated_copy=False):
+    def _get_custom_dedup_cli_volume(self, support_deduplicated_copy, with_deduplicated_copy, name='source_volume',
+                                     pool_name='pool_name'):
+        volume = self._get_cli_volume(with_deduplicated_copy, name=name, pool_name=pool_name)
+        if not support_deduplicated_copy:
+            del volume.deduplicated_copy
+        return volume
+
+    def _prepare_mocks_for_create_snapshot(self, support_deduplicated_copy=True, source_has_deduplicated_copy=False,
```

Comment:
are the part you added are relevant for all (or at least "most") create_snapshot tests?